### ISSUE TITLE
Update Django to version 4.2

### ISSUE THREAD

**Issue by tylerecouture** [bytedeck/bytedeck#1691]
Django 3.2 has reached end of life.  We need to update to a supported version.  Then next LTS is 4.2, so let's start with that Upgrade.  5.2 LTS won't be released until mid 2025.

Step 1 Clean up other dependencies:

To make life easier, before upgrading Django (which may create other incompatibilities with our current dependency versions), please simplify our requirements.txt files.  Currently they are a bit of a mess, and include almost all of our sub-dependencies, with inconsistent version pinning.

- [ ] Use `pipdeptree` (run from command line within venv) to view the root dependacies, and their sub dependancies.  Ideally we only have our root dependancies in our requirements files, and rely on those to properly select the version of subdependancies they need.  So remove all subdependancies. 
- [ ] Sometimes, two root dependancies rely on the same sub-dependancy and we have to pin the sub-dependancy due to make sure they are both compatible.  if so, make sure to leave a comment in the file. 
- [ ] All dependancies should by default be pinned like this: >=current.version.bugfix, <next.version (i.e. allow for bugfix updates if the dependancy uses semantic versioning).  Any dependancies that are pinned differently than this should have a comment explaining why.
- [ ] Submit this requirements update (removal of subdependancies) as its own PR before starting on the django version upgrade.

Step 2 Some Docs:
Some notes to review before starting:

- Django supported versions and timeline: https://www.djangoproject.com/download/#supported-versions
- Django Release Process: https://docs.djangoproject.com/en/dev/internals/release-process/

Step 3 Upgrade:
Read the backwards incompatibilities and upgrade guide here (along with anything else that is relevant)
 https://docs.djangoproject.com/en/5.1/releases/4.2/ includes:
